fix liveController processes status check

lilachmaliniak opened the pull request.

jessp01 commented on serviceWrappers/linux/kLiveController:
```diff
@@ -93,22 +97,18 @@ restart() {
-	        RETVAL=$(( $RETVAL + 1 ))
-        fi
+            status -p ${PIDFILE}
+            if [ ! $? -eq 0 ]; then
```
```
if [ ! $? -eq 0 ]; then
```
should be:
```
if [ $? -ne 0 ]; then
```
Also, as I wrote in https://kaltura.atlassian.net/browse/PLAT-7392
However, there's another problem here: you're checking the status of two different PIDs in the same function [in a for loop] but you do not return in the event status -p /var/run/kLiveController_0.pid /opt/kaltura/liveController/latest/lib/App.js fails, as a result, there may be a case where /var/run/kLiveController_0.pid does not contain a valid PID but /var/run/kLiveController_1.pid does and the status action will therefore return with RC 0, even though one of the two procs [the first one] is not running.
In the event /var/run/kLiveController_0.pid does not contain a valid PID, you should return with $?.

jessp01 commented on serviceWrappers/linux/kLiveController:
```diff
@@ -93,28 +105,28 @@ restart() {
+	            return ${RETVAL}
+	        fi
+	    else
+	        echo "${PIDFILE} is empty, ${NAME} is not running (verify by: ps aux | grep node)"
```
need to properly indent, also, the message is wrong. This point in the code will only be reached if PIDFILE is not readable, not if it's empty.

lilachmaliniak replied:
updated echo message

jessp01 commented on serviceWrappers/linux/kLiveController:
```diff
@@ -32,18 +37,18 @@ HOME=${HOME:-~}
-    [ -z "$NVM_DIR" ] && NVM_DIR="$HOME/.nvm"
-    if [ -s "$NVM_DIR/nvm.sh" ] ; then
-        source "$NVM_DIR/nvm.sh"
+    cd ${KLIVE_CONTROLLER_PREFIX}
```
Why is cd needed here? if you just need to check that ${KLIVE_CONTROLLER_PREFIX}/.nvmrc isn't empty then use the full path and check.

lilachmaliniak replied:
changed

jessp01 commented on serviceWrappers/linux/kLiveController:
```diff
@@ -53,20 +58,20 @@ start() {
+        PIDFILE="${PID_DIR}/${NAME}_${prefix}.pid"
+        LOGFILE="${LOG_DIR}/${NAME}_${prefix}-forever.log"
+        if [ -f ${PIDFILE} ]; then
             echo "${NAME}_${prefix} already running"
```
call status. Just because there is a PID file does not mean we are running.

lilachmaliniak replied:
corrected

lilachmaliniak commented:
updated kLiveController and liveRecorder scripts
highlights:
- use functions script to get process status
- call . /opt/rh/python27/enable to add python path to PATH 
- removed kill (called from stop if needed)
usage kLiveController {start|stop|restart|status}

kalturaguy commented on liveRecorder/serviceWrappers/linux/liveRecorder:
```diff
@@ -18,22 +18,29 @@
-LOGFILE="$LOG_DIR/${NAME}-service.log"
+PIDFILE="${PID_DIR}/${NAME}.pid"
+LOGFILE="${LOG_DIR}/${NAME}-service.log"
+LOGFILE="${LOG_DIR}/${NAME}-service.log"
```
duplicate line

lilachmaliniak replied:
done

jessp01 commented on liveRecorder/serviceWrappers/linux/liveRecorder:
```diff
@@ -44,30 +51,41 @@ createFolders() {
 
 stop() {
-    if [ -f $PIDFILE ]; then
+    if [ -f ${PIDFILE} ]; then
```
Again, you need to check that the PID is really alive, the fact the PID file is present is not enough.
Call status.

lilachmaliniak replied:
done

jessp01 commented on serviceWrappers/linux/kLiveController:
```diff
@@ -32,18 +37,17 @@ HOME=${HOME:-~}
             nvm use || nvm install
         else
-            echo "no .nvmrc in $PWD , Exiting"
+            echo "no .nvmrc in ${PWD} , Exiting"
```
Change this to no " ${KLIVE_CONTROLLER_PREFIX}/.nvmr", CWD should never be relied upon.

lilachmaliniak replied:
done

jessp01 commented on serviceWrappers/linux/kLiveController:
```diff
@@ -79,11 +84,18 @@ stop() {
         if [ -f $PIDFILE ]; then
```
Same story, use status to verify the PID in PIDFILE is a valid one.

lilachmaliniak replied:
done
